File: src/main/java/me/oriharel/machinery/utilities/BiSupplier.java
package me.oriharel.machinery.utilities;

/**
 * Callback with two parameters
 * @param <T> parameter one type
 * @param <U> parameter two type
 */
@FunctionalInterface
public interface BiSupplier<T, U> {
    void apply(T param1, U param2);
}

File: src/main/java/me/oriharel/machinery/utilities/BiSupplier.kt
package me.oriharel.machinery.utilities

/**
 * Callback with two parameters
 * @param <T> parameter one type
 * @param <U> parameter two type
</U></T> */
@FunctionalInterface
interface BiSupplier<T, U> {
    fun apply(param1: T, param2: U)
}